DataCollector/TwigCacheCollector.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace EmanueleMinotto\TwigCacheBundle\DataCollector;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollectorInterface;
+
+class TwigCacheCollector implements DataCollectorInterface
+{
+    /**
+     * @var string
+     */
+    private $strategyClass;
+
+    /**
+     * Data about fetchBlock requests.
+     *
+     * @var array
+     */
+    private $fetchBlock = [];
+
+    /**
+     * Data about generateKey requests.
+     *
+     * @var array
+     */
+    private $generateKey = [];
+
+    /**
+     * Cache hits.
+     *
+     * @var int
+     */
+    private $hits = 0;
+
+    public function collect(Request $request, Response $response, \Exception $exception = null)
+    {
+        // nothing to do here
+    }
+
+    /**
+     * @return string The collector name
+     */
+    public function getName()
+    {
+        return 'asm89_cache';
+    }
+
+    /**
+     * @param string $strategyClass
+     */
+    public function setStrategyClass($strategyClass)
+    {
+        $this->strategyClass = $strategyClass;
+    }
+
+    /**
+     * Store a fetch request.
+     *
+     * @param mixed  $key
+     * @param string $output
+     */
+    public function addFetchBlock($key, $output)
+    {
+        $this->fetchBlock[] = [$key, $output];
+
+        if ($output) {
```

Comment:
is this a string or a boolean?

Reply:
I don't think so - but this isn't new code. It's from https://github.com/EmanueleMinotto/TwigCacheBundle/pull/22/files/f87e43fa0593ad175a0e1100c994841d4d48c6df#diff-0c66391443147223f158b428590c98bfL79

Reply:
So, it's a string? Or it's a boolean?

Reply:
Looks like it is a string. But as I already mentioned. I am not creating a new feature - I am splitting the collector out of the twig extension. The code is still the same as before.

Reply:
@garak its arrayOrNull